client-v2/src/shared/components/input/InputImage.tsx

```diff
@@ -83,10 +85,68 @@ class InputImage extends React.Component<Props, State> {
   };
   public clearField = () => this.props.input.onChange(null);
 
+  public validMessage(data: GridData) {
+    return data &&
+           data.crop &&
+           data.crop.data &&
+           data.image &&
+           data.image.data;
+  }
+
+  public onMessage = (event: MessageEvent) => {
+    if (event.origin !== this.props.gridUrl) {
+      // Log: did not come from the grid
+      return;
+    }
+
+    const data: GridData = event.data;
+
+    if (!data) {
+      // TODO Log did not get data
```

Comment:
For future, but we have quite a few @todos across the codebase that relate to logging -- perhaps we can send errors directly to Sentry in these places